CODINGSTYLE.md

```diff
@@ -0,0 +1,490 @@
+
+	case 1:
+		DoSomething();
+		/* FALL THROUGH */
+
+	case 2:
+		DoMore();
+		b = a;
+		break;
+
+	case 3: {
+		int r = 2;
+
+		DoEvenMore(a);
+		/* FALL THROUGH */
+	}
+
+	case 4: {
+		int q = 345;
+
+		DoIt();
+		break;
+	}
+
+	default:
+		NOT_REACHED();
+}
+
+for (int i = 0; i < 10; i++) {
+	Foo();
+	Bar();
+}
+```
+
+### Classes
+* Classes names also use CamelCase.
+* Classes should have "public", "protected", and "private" sections.
+* Within these section the order is: types, static const members, static members, members, constructors / destructors, static methods, methods.
+* Deviations from above order are allowed when the code dictates it (e.g. a static const is needed for a typedef)
+* Methods and members ought to be grouped logically.
+* All those sorting rules sometimes conflict which one another. Please use common sense what increases legibility of the code in such a case.
+* The method implementation should indicate if it is virtual or similar by using a comment.
+* Very short methods can have one-line definition (if defined in the class scope).
+```c++
+class ThisIsAClass {
+public:
+	typedef Titem_   *ItemPtr;
+private:
+	static const int MAX_SIZE = 500;
+	int              size;
+	ItemPtr          *items;
+
+public:
+	explicit ThisIsAClass();
+	~ThisIsAClass();
+
+	int GetSize() { return this->size; }
+	virtual void Method();
+};
+
+/*virtual*/ void Class::Method()
+{
+	this->size *= 2;
+}
+```
+
+### Templates
+Templates are a very powerful C++ tool, but they can easily confuse beginners. Thus:
+* Templates are to be documented in a very clear and verbose manner. Never assume anything in the documentation.
+* the template keyword and the template layout get a separate line. typenames are either "T" or preceded by a "T", integers get a single capital letter or a descriptive name preceded by "T".
+```c++
+template <typename T, typename Tsomething, int N, byte Tnumber_of_something>
+int Func();
+```
+
+* If you are writing one or more template class in the dedicated header file, use file.hpp for its name instead of file.h. This will let others know that it is template library (includes also implementation), not just header with declarations.
+
+### Other important rules
+* Put a space before and after binary operators: "a + b", "a == b", "a & b", "a <<= b", etc.. Exceptions are ".", "->" and "[]" (no spaces) and "," (just space after it).
+* Put parenthesis where it improves readability: "*(b++)" instead of "*b++", and "if ((a & b) && c == 2)" instead of "if (a & b && c == 2)".
+* Do not put external declarations in implementation (i.e. cpp) files.
+* Use const where possible.
+* Do not typedef enums and structs.
+* Don't treat non-flags as flags: use "if (char_pointer != NULL && *char_pointer != '\0')", not "if (char_pointer && *char_pointer)".
+* Use "free(p)" instead of "if (p != NULL) free(p)". "free(NULL)" doesn't hurt anyone.
+* No trailing whitespace. The svn server will not allow tabs or space on the end of lines.
```

Comment:
We've not used svn for a long time, though one of the github workflows checks for trailing whitespace and some situations of wrong tabs.